src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/Components/StreamFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Server.Kestrel.Http;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Infrastructure
+{
+    public class StreamFactory : DefaultComponentFactory<Streams>
```

Comment:
I'm not sure IComponentFactory needs a MaxPooled property. Certainly not a settable one. That could be a property of your non-default PoolingComponentFactory or whatever that ends up being called.

I'm also unsure about the need for the IComponentFactory.Reset method, since Frame(OfT) can easily reset it's own components. I don't think the IComponent interface needs to exist at all. Once again, Frame is best to decide when to reset/uninitialize components before it reuses/returns them. The settable CorrelationId seems like a super crazy and leaky abstraction. There has to be a better way to reduce contention.